x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealmSettingsTests.java

```diff
@@ -0,0 +1,443 @@
+            });
+            assertThat(
+                exception.getMessage(),
+                equalTo("Invalid value [" + rejectedValue + "] for [" + settingKey + "]." + " Allowed values are [sharedsecret, none]}].")
+            );
+        }
+        for (final String ignoredValue : new String[] { null }) {
+            final Settings settings = Settings.builder().put(settingKey, ignoredValue).build();
+            final String actualValue = this.buildConfig(settings).getSetting(setting);
+            assertThat(actualValue, equalTo(setting.getDefault(settings)));
+        }
+        for (final String acceptedValue : new String[] { "sharedsecret", "none" }) {
+            final Settings settings = Settings.builder().put(settingKey, acceptedValue).build();
+            final String actualValue = this.buildConfig(settings).getSetting(setting);
+            assertThat(actualValue, equalTo(acceptedValue));
+        }
+    }
+
+    public void testAuthorizationRealms() {
+        final Setting.AffixSetting<List<String>> setting = DelegatedAuthorizationSettings.AUTHZ_REALMS.apply(JwtRealmSettings.TYPE);
+        final String settingKey = RealmSettings.getFullSettingKey(REALM_NAME, setting);
+        for (final String ignoredValue : new String[] { null, "" }) {
+            final Settings settings = Settings.builder().put(settingKey, ignoredValue).build();
+            final List<String> actualValue = this.buildConfig(settings).getSetting(setting);
+            assertThat(actualValue, equalTo(List.of()));
+        }
+        for (final String acceptedValue : new String[] { "a", "1", "native1,file1,ldap1,ad1" }) {
+            final Settings settings = Settings.builder().put(settingKey, acceptedValue).build();
+            final List<String> actualValue = this.buildConfig(settings).getSetting(setting);
+            assertThat(actualValue, equalTo(Arrays.asList(acceptedValue.split(",", -1))));
+        }
+    }
+
+    public void testSecureStrings() {
+        for (final Setting.AffixSetting<SecureString> setting : List.of(
+            JwtRealmSettings.ISSUER_HMAC_SECRET_KEY,
+            JwtRealmSettings.CLIENT_AUTHENTICATION_SHARED_SECRET
+        )) {
+            final String settingKey = RealmSettings.getFullSettingKey(REALM_NAME, setting);
+            for (final String rejectedValue : new String[] { null }) {
+                final Exception exception = expectThrows(NullPointerException.class, () -> {
+                    final MockSecureSettings secureSettings = new MockSecureSettings();
+                    secureSettings.setString(settingKey, rejectedValue);
+                    final SecureString actualValue = secureSettings.getString(settingKey);
+                    fail("No exception. Expected one for " + settingKey + "=" + rejectedValue + ". Got " + actualValue + ".");
+                });
+                assertThat(
```

Comment:
This seems like a very artificial scenario to test. KeyStoreWrapper.setString() wouldn't allow you to put a null value in there in the first place. In any case, the exception is thrown by KeyStoreWrapper.getString() and if that doesn't hand;e the null case and it should, then that is where we would be adding the test, I think

Reply:
Same situation as the yaml parsing.

This unit test is about directly testing the ES settings parsers. Config settings have not been preformatted by a YAML parsing or a KeyStoreWrapper. These are direct tests without any preconditioning of the input values.